src/schemas/compositions/usePrefectKind.ts

```diff
@@ -0,0 +1,34 @@
+import { Ref, computed } from 'vue'
+import { PrefectKind, isPrefectKindValue } from '@/schemas/types/schemaValues'
+
+export type UsePrefectKind = {
+  kind: Ref<PrefectKind>,
+}
+
+export function usePrefectKind(propertyValue: Ref<unknown>): UsePrefectKind {
+  const kindValuesMap: Partial<Record<PrefectKind, unknown>> = {}
+
+  const kind = computed<PrefectKind>({
+    get() {
+      if (isPrefectKindValue(propertyValue.value)) {
+        return propertyValue.value.__prefect_kind
+      }
+
+      return 'none'
+    },
+    set(__prefect_kind) {
+      kindValuesMap[kind.value] = propertyValue.value
+
+      if (__prefect_kind in kindValuesMap) {
+        propertyValue.value = kindValuesMap[__prefect_kind]
+        return
+      }
+
+      propertyValue.value = { __prefect_kind }
```

Comment:
Making sure I understand this correctly - this is what stores the value for a given kind view when switching between them, right?

Reply:
Correct, this composition handles caching values when switching and setting the correct value when the user changes the `kind`. Implementation components don't need to care about the value when switching kinds.